<a href="https://colab.research.google.com/github/MaryamNaveed-bioinfo/Flyranks-internship-assignmnet-1/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane or say 'freestyle' and describe your own question. One short paragraph: why this one?*

#My lane: Refresh / Content Opportunity Scoring

I'm choosing this lane because it builds on an idea I've already explored hands-on: using a
trained model to rank pages by risk, instead of relying only on a fixed hand-written rule.
I saw firsthand that a simple rule can be strong at the very top of a ranking, but a model
can outperform it deeper into the list where a rule runs out of signal. I want to build a
system that helps a reviewer quickly identify which pages, out of thousands, deserve
attention first instead of manually checking every page or relying on gut feeling.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision: Which pages should a content reviewer look at first, given limited time and thousands of pages to choose from?
Unit of analysis: One page (one row = one content item).
Output: A ranked list of pages, ordered by priority score highest-priority pages
at the top.

Action: A reviewer could refresh, expand, protect, prune, or simply monitor a flagged page, depending on the reason code attached to it.

Cost of a wrong call: If the model wrongly flags a healthy page as "needs review," the reviewer wastes limited time checking a page that didn't actually need attention.
If it misses a truly declining page, that page keeps losing traffic unnoticed a missed
opportunity, though not catastrophic, since a human still reviews before any real action
is taken.

Why data/ML helps at all:With thousands of pages and limited reviewer time, manually checking every page isn't realistic.
A ranked, evidence-based shortlist lets a small team
focus their limited attention where it's statistically most likely to matter.

In [ ]:
import os, subprocess

if not os.path.isdir("Flyranks-internship-assignmnet-1"):
    subprocess.run(["git", "clone", "--depth", "1",
        "https://github.com/MaryamNaveed-bioinfo/Flyranks-internship-assignmnet-1"], check=True)
os.chdir("Flyranks-internship-assignmnet-1")
print("Working dir:", os.getcwd())

Working dir: /content/Flyranks-internship-assignmnet-1


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# Number 1: how common is decline?
declining_rate = df["trend_direction"].str.lower().eq("down").mean()
print(f"{declining_rate:.1%} of pages are currently declining")

# Number 2: how many pages are stale but still get real traffic?
stale_visible = df[(df["days_since_last_update"] >= 180) & (df["impressions_90d"] >= 500)]
print(f"{len(stale_visible):,} pages are stale (180+ days unedited) but still get 500+ impressions")

# Number 3: how much traffic is tied up in declining pages?
declining_impressions = df[df["trend_direction"].str.lower() == "down"]["impressions_90d"].sum()
print(f"Declining pages account for {declining_impressions:,.0f} total impressions over 90 days")
print(df["impressions_90d"].describe())

54.2% of pages are currently declining
17 pages are stale (180+ days unedited) but still get 500+ impressions
Declining pages account for 79,994,363 total impressions over 90 days
count     30000.000000
mean       5200.366300
std       16838.019547
min           1.000000
25%          81.000000
50%         731.000000
75%        3615.250000
max      517715.000000
Name: impressions_90d, dtype: float64


These numbers show that decline is not a rare edge case: 54.2% of pages in the starter
dataset are currently declining. A smaller set of 17 pages are both stale (unedited for
180+ days) and still receiving significant traffic (500+ impressions) — clear candidates
for review. Declining pages collectively account for roughly 80 million impressions over
90 days, though this total is concentrated in a smaller number of high-traffic pages
(the median page gets only 731 impressions, while the top pages get over 500,000) — so a
review process should weight both how many pages are affected and how much traffic is
actually at stake per page, not treat all declining pages as equal priority.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This work can say: what patterns are *observed* in the starter data, which pages *appear*
stale or declining based on real signals, and *directional* evidence about where reviewer
attention may be most valuable. It is decision-support a ranked shortlist for a human to
review not a guarantee.

This work cannot say: that refreshing a flagged page will *cause* its traffic to recover,
that any signal in this data reflects an actual Google ranking factor, or that a page not
flagged is definitely safe. These are observed patterns in past data, not proof of what
will happen next or why.

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under work/notebooks/ — then submit your repo URL on the card. Done.